test/utils/test_patching.py

```diff
@@ -0,0 +1,239 @@
+@pytest.mark.parametrize("device", ["cuda:0", "cpu"])
+def test_image_fuse_with_multiple_batches(pytestconfig, device):
+    from modulus.utils.patching import image_batching, image_fuse
+
+    # Test with multiple batches
+    batch_size = 2
+
+    # Test cases: (img_shape_y, img_shape_x, patch_shape_y, patch_shape_x, overlap_pix, boundary_pix)
+    test_cases = [
+        (32, 32, 16, 16, 0, 0),  # Square image, no overlap/boundary
+        (64, 32, 32, 16, 0, 0),  # Rectangular image, no overlap/boundary
+        (48, 48, 16, 16, 4, 2),  # Square image, minimal overlap/boundary
+        (64, 48, 32, 16, 6, 2),  # Rectangular, larger overlap/boundary
+    ]
+
+    for (
+        img_shape_y,
+        img_shape_x,
+        patch_shape_y,
+        patch_shape_x,
+        overlap_pix,
+        boundary_pix,
+    ) in test_cases:
+        # Create original test image
+        original_image = (
+            torch.rand(batch_size, 3, img_shape_y, img_shape_x).to(device).float()
+        )
+
+        # Apply image_batching to split the image into patches
+        batched_images = image_batching(
+            original_image, patch_shape_y, patch_shape_x, overlap_pix, boundary_pix
+        )
+
+        # Apply image_fuse to reconstruct the image from patches
+        fused_image = image_fuse(
+            batched_images,
+            img_shape_y,
+            img_shape_x,
+            batch_size,
+            overlap_pix,
+            boundary_pix,
+        )
+
+        # Verify that image_fuse reverses image_batching
+        assert torch.allclose(fused_image, original_image, atol=1e-5), (
+            f"Failed on {device}: img=({img_shape_y},{img_shape_x}), "
+            f"patch=({patch_shape_y},{patch_shape_x}), "
+            f"overlap={overlap_pix}, boundary={boundary_pix}"
+        )
+
+
+@import_or_fail("cftime")
+def test_image_batching_basic(pytestconfig):
+    from modulus.utils.patching import image_batching
+
+    # Test with no overlap, no boundary, no input_interp
+    batch_size = 1
+    patch_shape_x = patch_shape_y = 4
+    overlap_pix = 0
+    boundary_pix = 0
+
+    input_tensor = torch.arange(1, 17).view(1, 1, 4, 4).cuda().float()
+    batched_images = image_batching(
+        input_tensor,
+        patch_shape_y,
+        patch_shape_x,
+        overlap_pix,
+        boundary_pix,
+    )
+    assert batched_images.shape == (batch_size, 1, patch_shape_y, patch_shape_x)
+    expected_output = input_tensor
+    assert torch.allclose(
+        batched_images, expected_output, atol=1e-5
+    ), "Batched images do not match expected output."
+
+
+@import_or_fail("cftime")
+def test_image_batching_with_boundary(pytestconfig):
+    from modulus.utils.patching import image_batching
+
+    # Test with boundary pixels, no overlap, no input_interp
+    patch_shape_x = patch_shape_y = 6
+    overlap_pix = 0
+    boundary_pix = 1
+
+    input_tensor = torch.ones(1, 1, 4, 4).cuda().float()  # All ones for easy validation
```

Comment:
Similar comment about `torch.ones` here. I see that generally the tougher and more complete tests are handled by the "round-trip" batching->fuse tests, but I think we might as well also make these simpler ones a bit more robust